server/index.js

```diff
@@ -48,14 +48,33 @@ async function setupApp() {
 
   });
 
+  app.post('/users', async (req, res) => {
+    try {
+      const { name, email, password, dateOfBirth, notificationPreference, biography, agreeToTerms, subscribeToNewsLetter } = req.body;
+      const user = await pgClient.query('SELECT * FROM  users WHERE email = $1', [email]);
+      if(user.length !== 0){
```

Comment:
I think user always returning something (a Result object), so you gotta look into the the correct property (rows or rowCount):

![image](https://github.com/nbktechworld/full-stack-web-dev/assets/13649002/37022d7d-9084-48e8-b6be-d992bdd1b20e)


```suggestion
      if(user.rows.length !== 0){
```